### DISCUSSION TITLE
AvaloniaUI #+ ANGLE + OpenGL 4.6

### DISCUSSION THREAD

**Discussion by markusalbrecht-procam** [AvaloniaUI/Avalonia#13774] [General]
Hey,
we are currently re-developing our 2d/3D CAD software with AvaloniaUI. On 2D side we have no issues it works like a charm but wehn it comes to 3D we are not sure how to continue and what is possible. We are trying to add the latest OpenES version 3.1+, this is absolutely necessary due to the needed features of the latest ES versions. It seems that with the latest version in combination with ANGLE we can only add V3.0 ???? How to integrate  the latest OpenGL/OpenES version into Avalonia via ANGLE ?
BR
Markus

**Comment by maxkatz6** [comment]
Avalonia uses Angle from October 6 - https://github.com/AvaloniaUI/angle/commits/avalonia-fork-master/
Which has incomplete support for 3.1 - https://github.com/AvaloniaUI/angle/blob/avalonia-fork-master/doc/ES31StatusOnD3D11.md
The same status in on latest master branch as well - https://github.com/google/angle/blob/main/doc/ES31StatusOnD3D11.md

If you need any of missing features in AngleGL, you can use WGL rendering mode. But WGL is known to have poorer stability with some drivers.
